tests/xgboost/test_xgboost_autolog.py

```diff
@@ -329,3 +404,54 @@ def test_xgb_autolog_does_not_throw_if_importance_values_are_empty(bst_params, t
     model = xgb.train(bst_params, dataset)
 
     assert model.get_score(importance_type="weight") == {}
+
+
+@pytest.mark.large
+def test_xgb_autolog_continues_logging_even_if_signature_inference_fails(bst_params, tmpdir):
+    tmp_csv = tmpdir.join("data.csv")
+    tmp_csv.write("1,0.3,1.2\n")
+    tmp_csv.write("0,2.4,5.2\n")
+    tmp_csv.write("1,0.3,-1.2\n")
+
+    mlflow.xgboost.autolog(importance_types=[])
+
+    # signature and input example inference should fail here since the dataset is given
+    #   as a file path
+    dataset = xgb.DMatrix(tmp_csv.strpath + "?format=csv&label_column=0")
+
+    xgb.train(bst_params, dataset)
+    run = get_latest_run()
+    run_id = run.info.run_id
+    artifacts_dir = run.info.artifact_uri.replace("file://", "")
+    client = mlflow.tracking.MlflowClient()
+    artifacts = [x.path for x in client.list_artifacts(run_id, "model")]
+
+    ml_model_filename = "MLmodel"
+    assert os.path.join("model", ml_model_filename) in artifacts
+    ml_model_path = os.path.join(artifacts_dir, "model", ml_model_filename)
+
+    data = None
+    with open(ml_model_path, "r") as f:
+        data = yaml.load(f, Loader=yaml.FullLoader)
+
+    assert data is not None
+    assert "run_id" in data
+    assert "signature" not in data
+
+
+@pytest.mark.large
+def test_xgb_autolog_does_not_break_dmatrix_serialization(bst_params):
+    mlflow.xgboost.autolog()
+
+    # we cannot use dtrain fixture, as the dataset must be constructed
+    #   after the call to autolog() in order to get the input example
+    iris = datasets.load_iris()
+    X = pd.DataFrame(iris.data[:, :2], columns=iris.feature_names[:2])
+    y = iris.target
+    dataset = xgb.DMatrix(X, y)
+
+    xgb.train(bst_params, dataset)
+
+    dataset.save_binary("dataset_serialization_test")  # serialization should not throw
+    xgb.DMatrix("dataset_serialization_test")  # deserialization also should not throw
+    os.remove("dataset_serialization_test")
```

Comment:
Should we use `tmpdir` here because if we keyboard-interrupt right after `xgb.DMatrix(...)`, the `"dataset_serialization_test"` file remains in the project root directory?

Reply:
true... I changed it :)